### ISSUE TITLE
Feature request: precise TTL output control from the Open Ephys aquisition board

### ISSUE THREAD

**Issue by paulmthompson** [open-ephys/rhythm#13]
I currently use old custom software (https://github.com/paulmthompson/Intan.jl) to control the Intan version of the acquisition board to generate pulse trains. I had to do some verilog modifications (similar to what the RHS system uses) so that I could have the FPGA generate precisely timed TTLs with custom frequency, pulse width etc.

Soon I am going to be using a different rig with a Open Ephys acquisition board.  I believe OpenEphys currently lacks this functionality, correct? Consequently, I am going to have to adapt my previous custom verilog code to use the OpenEphys board.

Would you be interested if I made a pull request of this feature? I can do the verilog modifications here in this repository and generate the bitfile but I might need some help making the OpenEphys plugin.

**Comment by jsiegle** [comment]
This sounds like it would be super useful! We actually have a [soon-to-be-released plugin](https://github.com/open-ephys-plugins/rhythm-plugins/blob/main/Source/AcqBoardOutput.h) for triggering TTL outputs on the acquisition board. Right now it only supports single pulses, with the pulse width determined in software, but it would be awesome to be able to define custom train parameters that are executed by the hardware.

I would say go ahead and submit a pull request, and @aacuevas can review it. If everything is working well on the hardware side, I can help make the necessary changes in software.

**Comment by paulmthompson** [comment]
Sounds good. I will take a look and make a pull request.

For reference, these are the files I had to change for the Intan board a few years ago:
https://github.com/paulmthompson/intan-fpga

The main functionality comes from the digout_sequencer.v file. Intan uses this to create the pulse sequences for the RHS stimulation headstages (and their digital outputs may do the same with that system), so I modified it to control the digital outputs on their regular fpga board. The parameters to describe the pulse trains is in the Intan guide at figure 32:
https://intantech.com/files/Intan_Stim_Recording_Controller_user_guide.pdf

![image](https://user-images.githubusercontent.com/8922014/171721197-552c0427-0d8a-42cb-983a-20df4949086e.png)
